connectors/amt_enhetsregister/src/main/kotlin/no/nav/amt/tiltak/connectors/amt_enhetsregister/AmtEnhetsregisterConnector.kt

```diff
@@ -0,0 +1,43 @@
+package no.nav.amt.tiltak.connectors.amt_enhetsregister
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+import no.nav.amt.tiltak.core.domain.enhetsregister.Virksomhet
+import no.nav.amt.tiltak.core.port.EnhetsregisterConnector
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import java.util.function.Supplier
+
+class AmtEnhetsregisterConnector(
+	private val url: String,
+	private val tokenProvider: Supplier<String>,
+	private val httpClient: OkHttpClient = OkHttpClient(),
+	private val objectMapper: ObjectMapper = ObjectMapper().registerKotlinModule(),
+) : EnhetsregisterConnector {
+
+	override fun hentVirksomhet(organisasjonsnummer: String): Virksomhet {
+		val request = Request.Builder()
+			.url("$url/api/enhet/$organisasjonsnummer")
+			.addHeader("Authorization", "Bearer ${tokenProvider.get()}")
+			.get()
+			.build()
+
+		httpClient.newCall(request).execute().use { response ->
+			if (!response.isSuccessful) {
+				throw RuntimeException("Klarte ikke å hente enhet fra amt-enhetsregister. Status: ${response.code}")
+			}
+
+			val body = response.body?.string() ?: throw RuntimeException("Body is missing")
+
+			val enhetDto = objectMapper.readValue(body, EnhetDto::class.java)
+
+			return Virksomhet(
+				navn = enhetDto.navn,
```

Comment:
En spesiell grunn til at DTOen heter Enhet mens i modellen vår så heter det VIrksomhet? Ville kanskje kalt det for Virksomhet og VirksomhetsDTO for å lettere se koblingen. Virker også litt rart at objektet virksomhet som har organisasjonsnummer isteden for virksomhetsnummer

Reply:
Det er litt vanskelig å vite hva som er riktig navn for dette. Brønnøysund kaller det "enhet", men uten kontekst så er det ikke så veldig tydlig hva "enhet" refererer til.

Virksomhet er tydligere, men jeg vet ikke om virksomhet er mer spisset enn enhet. 
Dvs at ikke alle enheter er virksomheter, mens alle enheter er også virksomheter